docs/modules/sql/pages/functions-and-operators.adoc

```diff
@@ -0,0 +1,713 @@
+
+|MAX(any) :: same type as the input
+|Calculates the maximum of the non-null input values. Applicable also to `OBJECT` type, if the underlying value is `java.lang.Comparable`.
+
+|===
+
+For examples of how to use aggregate functions, see the xref:select.adoc[`SELECT` statement documentation].
+
+== Conversion Functions
+
+Conversion functions allow you to convert the result type of one expression to another explicit type.
+
+.Descriptions of conversion functions
+[cols="1m,2,1,1"]
+|===
+| Function| Description| Example| Result
+
+|CAST(expression AS data_type)
+|Converts the result type of `expression` to `data_type`.
+|
+
+`SELECT CAST("2020-06-02 17:00:53.110+00:00" AS TIMESTAMP)``
+
+|
+
+`2020-06-03T00:00:53.11Z`
+|===
+
+== Date and Time Functions
+
+.Descriptions of date functions
+[cols="20%a,40%a,20%a,20%a"]
+|===
+| Function| Description| Example| Result
+
+|`EXTRACT(element, date)`
+|Returns the `element` date part from the `date` expression.
+
+Supported elements: `MILLENIUM`, `CENTURY`, `DECADE`, `YEAR`, `ISOYEAR`, `QUARTER`, `MONTH`, `WEEK`, `DAY`, `DOW`, `ISODOW`, `DOY`, `HOUR`, `MINUTE`, `SECOND`, `MILLISECOND`, `MICROSECOND`, `EPOCH`
+
+Supported date types: `DATE`, `TIME`, `TIMESTAMP`, `TIMESTAMP WITH TIME ZONE`
+|`EXTRACT(WEEK FROM "2017-06-15")`
+|24
+
+|`TO_TIMESTAMP_TZ(BIGINT)`
+|Converts a `BIGINT` value in seconds to `TIMESTAMP_WITH_TIMEZONE`.
+
+
+|`SELECT TO_TIMESTAMP_TZ(v) FROM TABLE (generate_series(1,3))`
+|
+1970-01-01T03:00:00.001+03:00
+
+1970-01-01T03:00:00.002+03:00
+
+1970-01-01T03:00:00.003+03:00
+
+|`TO_EPOCH_MILLIS(TIMESTAMP_WITH_TIMEZONE)`
+|Converts `TIMESTAMP_WITH_TIMEZONE` values to an `EPOCH` value in milliseconds.
+|`SELECT TO_EPOCH_MILLIS(TO_TIMESTAMP_TZ(v)) FROM TABLE (generate_series(1,3))`
+|1000
+
+2000
+
+3000
+|===
+
+== File Table Functions
+
+To execute an ad-hoc query against data in files you can use one of the following table functions:
+
+* `csv_file`
+* `json_flat_file`
+* `avro_file`
+* `parquet_file`
+
+File table functions create a temporary mapping to a file, which is valid for the duration
+of the statement. These functions accept the same options as those available for the xref:mapping-to-a-file-system.adoc[file connector].
+
+To configure the temporary mapping, you can use either positional arguments or named arguments:
+
+.Positional arguments in a file table function
+```sql
+SELECT * FROM TABLE(
+  CSV_FILE('/path/to/directory', '*.csv', MAP['key', 'value'])
+);
+```
+
+.Named arguments in a file table function
+```sql
+SELECT * FROM TABLE(
+  CSV_FILE(path => '/path/to/directory', options => MAP['key', 'value'])
+);
+```
+
+== JSON Functions
+
+Hazelcast supports the following functions, which can retrieve JSON data.
+
+.Descriptions of JSON functions
+[cols="[cols="1,2,1,1"]
+|===
+| Function :: Returns| Description|Example|Result
+
+|`JSON_QUERY(jsonArg, jsonPath)` :: `JSON`
```

Comment:
Might be worth mentioning wrapper modes here, there's a couple of tests that demonstrate this functionality: 
https://github.com/hazelcast/hazelcast/blob/master/hazelcast-sql/src/test/java/com/hazelcast/jet/sql/impl/expression/json/JsonQueryFunctionIntegrationTest.java#L103